src/geometry/manifold/derivation_bundle.lean

```diff
@@ -0,0 +1,153 @@
+{H : Type*} [topological_space H] (I : model_with_corners 𝕜 E H)
+(M : Type*) [topological_space M] [charted_space H M] (n : with_top ℕ)
+
+open_locale manifold
+
+-- the following two instances prevent poorly understood type class inference timeout problems
+instance smooth_functions_algebra : algebra 𝕜 C^∞⟮I, M; 𝕜⟯ := by apply_instance
+instance smooth_functions_tower : is_scalar_tower 𝕜 C^∞⟮I, M; 𝕜⟯ C^∞⟮I, M; 𝕜⟯ := by apply_instance
+
+/-- Type synonym, introduced to put a different `has_scalar` action on `C^n⟮I, M; 𝕜⟯`
+which is defined as `f • r = f(x) * r`. -/
+@[nolint unused_arguments] def pointed_smooth_map (x : M) := C^n⟮I, M; 𝕜⟯
+
+localized "notation `C^` n `⟮` I `,` M `;` 𝕜 `⟯⟨` x `⟩` :=
+  pointed_smooth_map 𝕜 I M n x" in derivation
+
+variables {𝕜 M}
+
+namespace pointed_smooth_map
+
+instance {x : M} : has_coe_to_fun C^∞⟮I, M; 𝕜⟯⟨x⟩ := by { rw pointed_smooth_map, apply_instance }
+instance {x : M} : comm_ring C^∞⟮I, M; 𝕜⟯⟨x⟩ := by { rw pointed_smooth_map, apply_instance }
+instance {x : M} : algebra 𝕜 C^∞⟮I, M; 𝕜⟯⟨x⟩ := by { dunfold pointed_smooth_map, apply_instance }
+instance {x : M} : inhabited C^∞⟮I, M; 𝕜⟯⟨x⟩ := ⟨0⟩
+
+instance {x : M} : algebra C^∞⟮I, M; 𝕜⟯⟨x⟩ C^∞⟮I, M; 𝕜⟯ :=
+by { dunfold pointed_smooth_map, apply_instance }
+
+instance {x : M} : is_scalar_tower 𝕜 C^∞⟮I, M; 𝕜⟯⟨x⟩ C^∞⟮I, M; 𝕜⟯ :=
+by { dunfold pointed_smooth_map, apply_instance }
+
+/-- Evaluation at a point is a ring homomorphism. Same thing as writing manually
+`to_fun := λ f, f x`.-/
+def eval' (x : M) : C^∞⟮I, M; 𝕜⟯⟨x⟩ →+* 𝕜 :=
+(pi.eval_ring_hom _ x : (M → 𝕜) →+* 𝕜).comp smooth_map.coe_fn_ring_hom
+
+variable {I}
+
+/-- The above evaluation gives rise to an algebra structure of `C^∞⟮I, M; 𝕜⟯` on `𝕜`. -/
+instance eval_algebra {x : M} : algebra C^∞⟮I, M; 𝕜⟯⟨x⟩ 𝕜 := (eval' I x).to_algebra
+
+/-- With the above algebra structure evaluation is actually an algebra morphism. -/
+def eval (x : M) : C^∞⟮I, M; 𝕜⟯ →ₐ[C^∞⟮I, M; 𝕜⟯⟨x⟩] 𝕜 :=
+{ commutes' := λ k, rfl, ..eval' I x }
+
+lemma scalar_def (x : M) (f : C^∞⟮I, M; 𝕜⟯⟨x⟩) (k : 𝕜) : f • k = f x * k := rfl
+
+instance (x : M) : is_scalar_tower 𝕜 C^∞⟮I, M; 𝕜⟯⟨x⟩ 𝕜 :=
+{ smul_assoc := λ k f h, by { simp only [scalar_def, algebra.id.smul_eq_mul, smooth_map.coe_smul,
+  pi.smul_apply, mul_assoc]} }
+
+end pointed_smooth_map
+
+open_locale derivation
+
+/-- The derivations at a point of a manifold. Some regard this as a possible definition of the
+tangent space -/
+@[reducible] def point_derivation (x : M) := derivation 𝕜 (C^∞⟮I, M; 𝕜⟯⟨x⟩) 𝕜
+
+variable (M)
+
+/-- The total bundle of point derivations. -/
+def derivation_bundle := Σ x : M, point_derivation I x
+
+variables {I M}
+
+/-- The inclusion map of derivations at a point into the total bundle. -/
+def derivation_inclusion {x : M} (v : point_derivation I x) : derivation_bundle I M :=
+sigma.mk x v
+
+instance [inhabited M] : inhabited (derivation_bundle I M) :=
+⟨derivation_inclusion (0 : point_derivation I (default M))⟩
```

Comment:
I'm not sure I understand the point of these definitions - you don't use them anywhere, not even in the `left_invariant` stuff you had before. `derivation_bundle` is really just `sigma (point_derivation)`, and in general probably anything you want to refer to as an `x_bundle` is just `sigma x`.

I think it might make sense to leave adding this definition until the PR that adds a non-trivial statement about it - until then, perhaps just mention "The total bundled of point derivations can be obtained with `sigma (point_derivation I)`" in the module docstring.

Reply:
```suggestion
```